File: Easy/349-intersection-of-two-arrays/try-6.py
/*
 * Problem #349: Intersection of Two Arrays
 * Difficulty: Easy
 * Submission: Try 6
 * status: Accepted
 * Language: python3
 * Date: 06/12/2024, 15:06:00
 * Link: https://leetcode.com/problems/intersection-of-two-arrays/
 */

class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        h1=set()
        h2=set()
        l1=[]

        for i in nums1:
            h1.add(i)
        for i in nums2:
            h2.add(i)

        for i in h1:
            if i in h2:
                l1.append(i)

        return l1

        

File: Easy/350-intersection-of-two-arrays-ii/try-6.cpp
/*
 * Problem #350: Intersection of Two Arrays II
 * Difficulty: Easy
 * Submission: Try 6
 * status: Accepted
 * Language: cpp
 * Date: 04/12/2024, 21:52:49
 * Link: https://leetcode.com/problems/intersection-of-two-arrays-ii/
 */

class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        unordered_map<int,int> map1;
        vector<int> vec;

        for(int i :nums1){
            map1[i]++;
        }
        for(int i : nums2){
            if(map1[i]>0){
                vec.push_back(i);
                map1[i]--;
            }
        }
        return vec;
    }
};
